src/main/java/xyz/nucleoid/extras/mixin/PlayerManagerMixin.java

```diff
@@ -0,0 +1,74 @@
+package xyz.nucleoid.extras.mixin;
+
+import net.minecraft.network.Packet;
+import net.minecraft.network.packet.s2c.play.PlayerListS2CPacket;
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayNetworkHandler;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import xyz.nucleoid.extras.PlayerListHelper;
+
+import java.util.Map;
+import java.util.UUID;
+
+@Mixin(PlayerManager.class)
+public abstract class PlayerManagerMixin {
+
+    @Shadow @Nullable public abstract ServerPlayerEntity getPlayer(UUID uuid);
+
+    @Shadow @Final private Map<UUID, ServerPlayerEntity> playerMap;
+
+    @Redirect(method = "onPlayerConnect", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/PlayerManager;sendToAll(Lnet/minecraft/network/Packet;)V"))
+    private void nucleoid$sendToAll(PlayerManager playerManager, Packet<?> packet1) {
```

Comment:
Unimportant things: don't need the `nucleoid$` prefix in these redirects, afaik Mixin will make sure they are unique by itself. Also `packet1` is not an ideal name- can be avoided by removing the `whitePacket` declaration and renaming the existing parameter to that. It's always casted when accessed anyway.